airflow/jobs/local_task_job.py

```diff
@@ -203,6 +205,7 @@ def sigusr2_debug_handler(signum, frame):
                     )
         finally:
             self.on_kill()
+            return return_code
```

Comment:
I guess this return might be the reason why Segmentation Fault handler tests broken, and potentially others like SIGTERM (but their not raise any error or run in separate process).